internal/app/app.go

```diff
@@ -0,0 +1,159 @@
+// Package app exports a D-Bus object to apply proxy settings.
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/introspect"
+	log "github.com/sirupsen/logrus"
+	"github.com/ubuntu/decorate"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/proxy"
+)
+
+const dbusObjectPath = "/com/ubuntu/ProxyManager"
+const dbusInterface = "com.ubuntu.ProxyManager"
+
+// DryRun is a context key to indicate that we should not apply the proxy
+// settings. It is intended to be used in tests only.
+type DryRun struct{}
+
+type busObject struct {
+	ctx    context.Context
+	cancel context.CancelCauseFunc
+
+	mu sync.Mutex
+}
+
+type options struct {
+	timeout time.Duration
+	initCtx context.Context
+}
+type option func(*options)
+
+// App is the main application object.
+type App struct {
+	conn    *dbus.Conn
+	obj     *busObject
+	timeout time.Duration
+}
+
+// Apply is a function called via D-Bus to apply the system proxy settings.
+func (b *busObject) Apply(http, https, ftp, socks, no, mode string) *dbus.Error {
```

Comment:
Maybe something like `ApplyProxy` or `ApplySettings` (even `ApplyProxySettings` if you are brave enough)? `busObject.Apply` does not say much about the purpose of the function.